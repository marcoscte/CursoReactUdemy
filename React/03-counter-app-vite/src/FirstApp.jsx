//Fragmento explicito
// import React, { Fragment } from 'react';
// export const FirstApp = () => {
//   return (
//     <Fragment>
//     <h1>Hola</h1>
//     <p>subtitulo</p>
//     </Fragment>
//   )
// }

import React from 'react';
import PropTypes from 'prop-types'; //sirve para definir tipo a las propiedades que recibe el functional component


const newMessage = [1,2,3,4,5,6,7,8,9];
const objeto = {
  message : 'holamundo', //no valido para mandar directamente a html en el return
  name : 'perro'
};


const suma = (n1,n2)=>{ //Evitar poner funciones dentro del funcntional component
  return n1 + n2;
}

export const FirstApp = ({title, subtitle}) => {
//Es raro encontrar las props de manera explicita, es mas comun desestructurar

  return ( //fragmento implicito <></>
  //Siempre se debe retornar un nodo padre
  //se debe colocar expresion de js permitida que no sea un objeto 
    <>

    <h1 data-testid="test-title">{title}</h1>
    <h3>{subtitle}</h3>
    <h3>{subtitle}</h3>
    
    <p>eliud</p>
    </>
  )
}

FirstApp.propTypes = {
    title:PropTypes.string.isRequired,
    subtitle : PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  subtitle : 'No hay subtitulo'
}



