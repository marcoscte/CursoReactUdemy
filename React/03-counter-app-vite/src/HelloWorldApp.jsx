//rafc para generar snippet por default
import React from 'react'

//functional component componente basado en una funcion
export const HelloWorldApp = () => {
  return (
    <div>HelloWorldApp</div>
  )
}
