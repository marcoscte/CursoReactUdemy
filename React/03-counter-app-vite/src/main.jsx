import React from 'react'
import ReactDOM from 'react-dom/client'
import { FirstApp } from './FirstApp'
import {HelloWorldApp} from './HelloWorldApp'

import './global.css' //importamos estilos globales, automaticamente se aplican a todo
import { CounterApp } from './CounterApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value = {11}/> 
    {/* <FirstApp title="hola, soy Eliud!" subtitle = "1234"/>  */}
  </React.StrictMode>
)
