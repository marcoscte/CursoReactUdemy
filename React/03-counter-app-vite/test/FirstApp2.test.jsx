import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


describe('Pruebas en FirstApp.jsx', () => {

  const title = "Hola, soy Naruto!";
  const subTitle = "Hola perros";

  test('debe hacer match con el snapshot', () => {
    const { container } = render(<FirstApp title={title} subTitle={subTitle} />);
    expect(container).toMatchSnapshot();
  });

  test('debe mostrar el texto Hola, soy Naruto!', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();



  });


  test('debe mostrar el titulo en un h1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);



  });

  test('debe enviar el subtitulo enviado por props', () => {
    render(<FirstApp title={title} subtitle={subTitle}/>);
    expect(screen.getAllByText(subTitle).length).toBe(2);



  });


})