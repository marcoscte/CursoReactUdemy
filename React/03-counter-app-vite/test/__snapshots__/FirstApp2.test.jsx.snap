// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp.jsx debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Naruto!
  </h1>
  <h3>
    No hay subtitulo
  </h3>
  <h3>
    No hay subtitulo
  </h3>
  <p>
    eliud
  </p>
</div>
`;
